### ISSUE TITLE
OnnxImportException: Unexpected error while parsing layer 19 of type Conv. Index was outside the bounds of the array.

### ISSUE THREAD

**Issue by 6Kwecky6** [Unity-Technologies/barracuda-release#184]
Hello

I'm importing a pretrained CNN from PyTorch, but getting a similar error to [this guy](https://github.com/Unity-Technologies/barracuda-release/issues/95), except I'm not using LSTM. My model is only a standard CNN, so I might have done something wrong.

Barracuda version: 1.0.4
Python: 3.9.2
PyTorch: 1.7.1
Opset version: 9

Adding the output from `onnx.helper.printable_graph(onnx_model.graph)`, then the stack trace:
Model graph:
'graph torch-jit-export (\n  %X[FLOAT, 1x16x40]\n) initializers (\n  %47[INT64, 1]\n  %CNN_blocks.0.bias[FLOAT, 512]\n  %CNN_blocks.0.weight[FLOAT, 512x16x3]\n  %CNN_blocks.11.bias[FLOAT, 4096]\n  %CNN_blocks.11.weight[FLOAT, 4096x4096x3]\n  %CNN_blocks.2.bias[FLOAT, 512]\n  %CNN_blocks.2.weight[FLOAT, 512x512x3]\n  %CNN_blocks.5.bias[FLOAT, 1024]\n  %CNN_blocks.5.weight[FLOAT, 1024x512x3]\n  %CNN_blocks.7.bias[FLOAT, 1024]\n  %CNN_blocks.7.weight[FLOAT, 1024x1024x3]\n  %CNN_blocks.9.bias[FLOAT, 4096]\n  %CNN_blocks.9.weight[FLOAT, 4096x1024x3]\n  %FC_blocks.0.bias[FLOAT, 2048]\n  %FC_blocks.0.weight[FLOAT, 2048x40960]\n  %FC_blocks.3.bias[FLOAT, 2048]\n  %FC_blocks.3.weight[FLOAT, 2048x2048]\n  %FC_blocks.6.bias[FLOAT, 2]\n  %FC_blocks.6.weight[FLOAT, 2x2048]\n) {\n  %19 = Conv[dilations = [1], group = 1, kernel_shape = [3], pads = [1, 1], strides = [1]](%X, %CNN_blocks.0.weight, %CNN_blocks.0.bias)\n  %20 = Relu(%19)\n  %21 = Conv[dilations = [1], group = 1, kernel_shape = [3], pads = [1, 1], strides = [1]](%20, %CNN_blocks.2.weight, %CNN_blocks.2.bias)\n  %22 = Relu(%21)\n  %23 = MaxPool[kernel_shape = [2], pads = [0, 0], strides = [2]](%22)\n  %24 = Conv[dilations = [1], group = 1, kernel_shape = [3], pads = [1, 1], strides = [1]](%23, %CNN_blocks.5.weight, %CNN_blocks.5.bias)\n  %25 = Relu(%24)\n  %26 = Conv[dilations = [1], group = 1, kernel_shape = [3], pads = [1, 1], strides = [1]](%25, %CNN_blocks.7.weight, %CNN_blocks.7.bias)\n  %27 = Relu(%26)\n  %28 = Conv[dilations = [1], group = 1, kernel_shape = [3], pads = [1, 1], strides = [1]](%27, %CNN_blocks.9.weight, %CNN_blocks.9.bias)\n  %29 = Relu(%28)\n  %30 = Conv[dilations = [1], group = 1, kernel_shape = [3], pads = [1, 1], strides = [1]](%29, %CNN_blocks.11.weight, %CNN_blocks.11.bias)\n  %31 = Relu(%30)\n  %32 = MaxPool[kernel_shape = [2], pads = [0, 0], strides = [2]](%31)\n  %33 = Shape(%32)\n  %34 = Constant[value = <Scalar Tensor []>]()\n  %35 = Gather[axis = 0](%33, %34)\n  %37 = Unsqueeze[axes = [0]](%35)\n  %39 = Concat[axis = 0](%37, %47)\n  %40 = Reshape(%32, %39)\n  %41 = Gemm[alpha = 1, beta = 1, transB = 1](%40, %FC_blocks.0.weight, %FC_blocks.0.bias)\n  %42 = Relu(%41)\n  %43 = Gemm[alpha = 1, beta = 1, transB = 1](%42, %FC_blocks.3.weight, %FC_blocks.3.bias)\n  %44 = Relu(%43)\n  %45 = Gemm[alpha = 1, beta = 1, transB = 1](%44, %FC_blocks.6.weight, %FC_blocks.6.bias)\n  %Y = Softmax[axis = 1](%45)\n  return %Y\n}'

Stach trace:
OnnxImportException: Unexpected error while parsing layer 19 of type Conv.
Index was outside the bounds of the array.

Json: { "input": [ "X", "CNN_blocks.0.weight", "CNN_blocks.0.bias" ], "output": [ "19" ], "name": "Conv_0", "opType": "Conv", "attribute": [ { "name": "dilations", "ints": [ "1" ], "type": "INTS" }, { "name": "group", "i": "1", "type": "INT" }, { "name": "kernel_shape", "ints": [ "3" ], "type": "INTS" }, { "name": "pads", "ints": [ "1", "1" ], "type": "INTS" }, { "name": "strides", "ints": [ "1" ], "type": "INTS" } ] }
  at Unity.Barracuda.ONNXModelImporter.<.ctor>b__14_60 (Unity.Barracuda.ModelBuilder net, Unity.Barracuda.ONNXNodeWrapper node) [0x00041] in <ProjectPath>\Library\PackageCache\com.unity.barracuda@1.0.4\Barracuda\Editor\ONNXModelImporter.cs:640 
  at Unity.Barracuda.ONNXModelImporter.ConvertOnnxModel (Onnx.ModelProto onnxModel) [0x0032f] in <ProjectPath>\Library\PackageCache\com.unity.barracuda@1.0.4\Barracuda\Editor\ONNXModelImporter.cs:1088 

Unity.Barracuda.ONNXModelImporter.Err (Unity.Barracuda.Model model, System.String layerName, System.String message, System.String extendedMessage, System.String debugMessage) (at Library/PackageCache/com.unity.barracuda@1.0.4/Barracuda/Editor/ONNXModelImporter.cs:1404)
Unity.Barracuda.ONNXModelImporter.ConvertOnnxModel (Onnx.ModelProto onnxModel) (at Library/PackageCache/com.unity.barracuda@1.0.4/Barracuda/Editor/ONNXModelImporter.cs:1097)
Unity.Barracuda.ONNXModelImporter.OnImportAsset (UnityEditor.AssetImporters.AssetImportContext ctx) (at Library/PackageCache/com.unity.barracuda@1.0.4/Barracuda/Editor/ONNXModelImporter.cs:1005)
UnityEditor.AssetImporters.ScriptedImporter.GenerateAssetData (UnityEditor.AssetImporters.AssetImportContext ctx) (at <0b8db76a688c4d18a15dd7dfe092a035>:0)
UnityEditorInternal.InternalEditorUtility:ProjectWindowDrag(HierarchyProperty, Boolean)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&)

**Comment by AlexRibard** [comment]
Thank you for raising this issue!
Could you share the onnx file?

Thank you!

**Comment by 6Kwecky6** [comment]
Hey!
Thanks for a rapid answer!
The file is ~500MB, so it's unfortunately too large to be attached here. It might be a bit unwieldy to read the graph, and I exported it from PyTorch using same implementation that is being used in the PyTorch doc .
`torch.onnx.export(model,
                   x_torch,
                   'Models/CNN1D_M2.onnx',
                   export_params=True,
                    opset_version=9,
                  do_constant_folding=True,
                 input_names=['X'],
                 output_names=['Y'])`
If you still want the file, I can send it via mail or something.

**Comment by mantasp** [comment]
@6Kwecky6 you can upload your model to Google Drive or similar service and us the link to `barracuda-support@unity3d.com`

**Comment by FlorentGuinier** [comment]
For internal tracking purpose: 
Issue is being discussed in email thread _**"[Barracuda support] OnnxImportException: Unexpected error while parsing layer 19 of type Conv. Index was outside the bounds of the array."**_

**Comment by FlorentGuinier** [comment]
Closing as there is no activity on the thread (including email thread), please reopen if needed.
